test/index.test.ts

```diff
@@ -21,149 +23,187 @@ const testCases = [
+  it('should be prefixed with "\x19\x45thereum Signed Message:\n"', () => {
     const hashedMessage = eip191Signer.hashEthereumSignedMessage('Hello World');
 
     expect(hashedMessage).toBe(
-      '0x677739c1b99336b0c40ed12a4d77c68805f8b5ca2d865676de85bf83b3b664ee',
+      '0xa1de988600a42c4b4ab089b619297c17d53cffae5d5120d82d8a92d0bb3b78f2',
     );
   });
 });
 
-describe('Sign ethereum signed message function', () => {
-  testCases.forEach((data) => {
-    const signedObject = eip191Signer.signEthereumSignedMessage(data, signingKey);
+describe('function hashDataWithIntendedValidator', () => {
+  it('should be hexadecimal and of 32 bites', () => {
+    testCases.forEach((data) => {
+      const hashedData = eip191Signer.hashDataWithIntendedValidator(
+        validatorAddress,
+        data,
+      );
 
-    it('should return an object with the expected properties', () => {
-      expect(signedObject).toHaveProperty('message');
-      expect(signedObject).toHaveProperty('messageHash');
-      expect(signedObject).toHaveProperty('v');
-      expect(signedObject).toHaveProperty('r');
-      expect(signedObject).toHaveProperty('s');
-      expect(signedObject).toHaveProperty('signature');
+      expect(hashedData.substring(0, 2)).toBe('0x');
+      expect(hashedData.length).toBe(66);
     });
   });
-});
 
-describe('Recover the address function of an ethereum signed message', () => {
-  it('should recover the signing address when the message is not prefixed', () => {
-    testCases.forEach((data) => {
-      const messageData = eip191Signer.signEthereumSignedMessage(data, signingKey);
-      const message = messageData.message;
-      const signature = messageData.signature;
-      const recoveredAddress = eip191Signer.recover(message, signature);
+  it('should be prefixed with "\x19\x000xAd278a6eAd89f6B6c6Fdf54A3E6E876660593b45"', () => {
+    const hashedMessage = eip191Signer.hashDataWithIntendedValidator(
+      validatorAddress,
+      'Hello World',
+    );
 
-      expect(recoveredAddress).toBe(signingAddress);
-      expect(recoveredAddress.substring(0, 2)).toBe('0x');
-      expect(recoveredAddress.length).toBe(42);
-    });
+    expect(hashedMessage).toBe(
+      '0xa63022286ecaa3317625e319a64b3bf01c41da558dfc1890e8cb196eb414ffd5',
+    );
   });
 
-  it('should recover the signing address when recovering signed hash', () => {
+  it("should throw Error with message 'Validator needs to be a valid address' when validator's address is invalid", () => {
     testCases.forEach((data) => {
-      const messageData = eip191Signer.signEthereumSignedMessage(data, signingKey);
-      const message = messageData.message;
-      const signature = messageData.signature;
-
-      const messagePrefixed = eip191Signer.signEthereumSignedMessage(message);
-      const isMessagePrefixed = true;
+      const invalidValidatorAddress =
+        '0xC1912fEE45d61C87Cc5EA59DaE31190FFFFf232d';
```

Comment:
what makes this address invalid? Wrong checksum?

Maybe using smtgh like:

```
0xImANonValidAddress
```

would make it even more obvious?

Reply:
I disagree, checking for a wrong checksum, but valid hex is a better test :)